File: backend/interview_portal/src/main/java/by/interview/portal/dto/FullInterviewInfoDTO.java
package by.interview.portal.dto;

import by.interview.portal.domain.DisciplineMark;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class FullInterviewInfoDTO {

    private Long id;

    private String description;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    private CandidateBaseInfoDTO candidate;

    private String place;

    private Byte finalMark;

    private String feedback;

    private String status;

    private Set<UserBaseInfoDTO> interviewerSet;

    private Set<DisciplineBaseInfoDTO> disciplineSet;

    private List<DisciplineMark> markList;

}

File: frontend/src/app/api/models/full-interview-info-dto.ts
/* tslint:disable */
import { DisciplineBaseInfoDTO } from './discipline-base-info-dto';
import { UserBaseInfoDTO } from './user-base-info-dto';
import { DisciplineMark } from './discipline-mark';
import { CandidateDTO } from './candidate-dto';

export class FullInterviewInfoDTO {

  id?: number;

  candidate?: CandidateDTO;

  disciplineSet?: DisciplineBaseInfoDTO[];

  discipline?: DisciplineBaseInfoDTO;

  endTime?: string;

  feedback?: string;

  finalMark?: string;

  description?: string;

  interviewerSet?: UserBaseInfoDTO[];

  markList?: DisciplineMark[];

  place?: string;

  startTime?: string;

  status?: string;
}
